### PR TITLE
drivers:platform:maxim: I2C driver initial commit

### PR DESCRIPTION
**Pull request by adrimbarean**
Add MAX32660 I2C driver allowing for master mode write and read
transactions.

Signed-off-by: Andrei Drimbarean <Andrei.Drimbarean@analog.com>
Signed-off-by: Ciprian Regus <ciprian.regus@analog.com>

This PR should be merged after https://github.com/analogdevicesinc/no-OS/pull/1207

### CODE REVIEW

File: drivers/platform/maxim/max32660_i2c.c [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+#include <stdint.h>
+#include <stdlib.h>
+
+#include "i2c.h"
+#include "no-os/error.h"
+#include "no-os/i2c.h"
+#include "no-os/i2c_extra.h"
+
+/** Used to know how many instances are created */
+static uint32_t nb_created_desc[MXC_I2C_INSTANCES];
```
Comment (amiclaus) [inline drivers/platform/maxim/max32660_i2c.c:49]: Are these macros `MXC_I2C_INSTANCES, MXC_I2C1` etc coming from `mxc_config.h`?

File: drivers/platform/maxim/max32660_i2c.c [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+	if (extra_init->id >= MXC_I2C_INSTANCES)
+		return -EINVAL;
+
+	*desc = calloc(1, sizeof(**desc));
+	if (!(*desc))
+		return -ENOMEM;
+
+	max32660_i2c = calloc(1, sizeof(*max32660_i2c));
+	if (!max32660_i2c) {
+		free(*desc);
```
Comment (amiclaus) [inline drivers/platform/maxim/max32660_i2c.c:106]: conditional jumps should be implemented :)

File: drivers/platform/maxim/max32660_i2c.c [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+#include <stdint.h>
+#include <stdlib.h>
+
+#include "i2c.h"
+#include "no-os/error.h"
+#include "no-os/i2c.h"
+#include "no-os/i2c_extra.h"
```
Comment (amiclaus) [inline drivers/platform/maxim/max32660_i2c.c:46]: `maxim/i2c_extra.h`?

File: drivers/platform/maxim/max32660_i2c.c [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+/**
+ * @brief I2C0 interrupt handler.
+ * @param none
+ * @return void
+ */
+void I2C0_IRQHandler(void)
+{
+	I2C_Handler(MXC_I2C0);
+
+	return;
```
Comment (amiclaus) [inline drivers/platform/maxim/max32660_i2c.c:60]: Is it a common practice to add returns in the void functions?

**Review by amiclaus — COMMENTED** [review]
Some remark + questions on my side.

**Comment by adrimbarean** [comment]
V1:
 - remove extra line;
 - correctly mention folder in which files reside;
 - use conditional jumps to return from init;
 - remove extra return statements.

**Comment by adrimbarean** [comment]
V2:
 - remove `@param` that caused doxygen error.

File: drivers/platform/maxim/i2c_extra.h [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+#define I2C_EXTRA_H_
+
+#include "mxc_config.h"
+#include "i2c_regs.h"
+
+/**
+ * @struct max332660_i2c_extra_init_param
+ * @brief MAX32660 specific I2C initialization structure
+ */
+struct max332660_i2c_extra_init_param {
```
Comment (CiprianRegus) [inline drivers/platform/maxim/i2c_extra.h:50]: There is a typo in this struct's name. The chip is max32660.

**Comment by adrimbarean** [comment]
V3:
 - `max332660` to `max32660` typo correction.

**Comment by adrimbarean** [comment]
V4:
 - ready for merging if PR is accepted.

**Review by amiclaus — APPROVED** [review]

**Comment by adrimbarean** [comment]
V5:
 - remove redundant `id` parameter from the extra structure;
 - update according to new Maxim drivers API.

**Review by buha — APPROVED** [review]

**Pull request merged**